src/managers/decoders/VaultDecoder.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+import {BaseDecoder} from "src/managers/decoders/BaseDecoder.sol";
+
+contract VaultDecoder is BaseDecoder {
+    // @desc deposit into the ERC4626 vault
+    // @tag receiver:address:the address of the receiver of the vault tokens
+    function deposit(uint256, address receiver) external view virtual returns (bytes memory addressesFound) {
+        addressesFound = abi.encodePacked(receiver);
+    }
```

Comment:
Why is the `amount` not encoded? Sure it's related to the merkle tree idea but I'm not able to follow right now

Reply:
Ok... I understand.

What if, for example, I want to add a limited amount? Can I do it with this pattern?Should I add another method in the decoder to handle that?